src/hotspot/cpu/riscv/c1_MacroAssembler_riscv.cpp

```diff
@@ -321,7 +321,8 @@ void C1_MacroAssembler::build_frame(int framesize, int bang_size_in_bytes) {
   // first instruction with a jump. For this action to be legal we
   // must ensure that this first instruction is a J, JAL or NOP.
   // Make it a NOP.
-  nop();
+  IncompressibleRegion ir(this);  // keep the nop as 4 bytes for patching.
```

Comment:
I think you should move the `assert_alignment(pc())` in `C1_MacroAssembler::verified_entry` here before this nop. Seens this assertion was miss-placed by previous backport PR: https://github.com/openjdk/riscv-port-jdk11u/pull/40

Reply:
Thanks for the review, fixed.